provider/core/src/error.rs

```diff
@@ -203,13 +204,14 @@ impl DataError {
     ///
     /// If the "log_error_context" feature is enabled, this logs the whole request. Either way,
     /// it returns an error with the resource key portion of the request as context.
-    pub fn with_req(self, req: &DataRequest) -> Self {
+    #[cfg_attr(not(feature = "log_error_context"), allow(unused_variables))]
+    pub fn with_req(self, key: ResourceKey, req: &DataRequest) -> Self {
```

Comment:
I wanted `with_req` to require both arguments because they are both important as context for the errors.  This is a side-effect of splitting the key and the options into separate arguments.